themes/gatsby-theme-abstract-blog/src/index.js

```diff
@@ -0,0 +1,32 @@
+import * as React from "react"
```

Comment:
I would assume we'd want this, the fallback component and post to all be `.tsx` to support the TS flavors of the starters, no?

Reply:
On that same point, are there updates to the `publish` script needed with this PR?

Reply:
This is internal to the theme and doesn't really affect the starters; I'd rather publish this first version with the new API before making any more changes to these